### PR TITLE
Removed Redundant/unused code,  corrected print function

### PR DESCRIPTION
**Pull request by saikumar-ramavath**
corrected print function of topo sort
Previous algorithm works only for given inputs, made algorithm generic works for any inputs of graph.
Optimized look up in visited dictionary
Remove unused piece code

### **Describe your change:**

output of algorithm is incorrect
Remove unused  piece of code
Made code generic, previous code works only for that set of inputs
optimized look up in visited dictionary

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
*  [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by Jenil-S — APPROVED** [review]
Nice Work :+1: LGTM

**Comment by Jenil-S** [comment]
@onlinejudge95 I think you can check LGTM..

**Comment by saikumar-ramavath** [comment]
> Nice Work 👍 LGTM

Thanks! 👍

**Comment by saikumar-ramavath** [comment]
> @onlinejudge95 I think you can check LGTM..

sure 😊

**Comment by onlinejudge95** [comment]
Guys checks failing for black

**Comment by saikumar-ramavath** [comment]
I'm new to this, I didn't get that
Can some one please explain!

**Comment by saikumar-ramavath** [comment]
> @onlinejudge95 I think you can check LGTM..

I'm new to this,  Tests are failing
Can you please explain that!

**Comment by saikumar-ramavath** [comment]
> Guys checks failing for black

I'm new to this, I didn't get that
Can some one please explain!

**Comment by saikumar-ramavath** [comment]
> Nice Work 👍 LGTM

I'm new to this, I didn't get that
Can some one please explain!

**Comment by Prajwalprakash3722** [comment]
> > Guys checks failing for black
> 
> 
> 
> I'm new to this, I didn't get that
> 
> Can some one please explain!

install black by `pip install black`

then run your file through black by 
`black filename.py` black is basically a auto format rule used by this repo owner all codes must pass this format, if you have any more doubt please reply back ASAP ✌🏼

**Comment by saikumar-ramavath** [comment]
> > > Guys checks failing for black
> > 
> > 
> > I'm new to this, I didn't get that
> > Can some one please explain!
> 
> install black by `pip install black`
> 
> then run your file through black by
> `black filename.py` black is basically a auto format rule used by this repo owner all codes must pass this format, if you have any more doubt please reply back ASAP ✌🏼

T

> > > Guys checks failing for black
> > 
> > 
> > I'm new to this, I didn't get that
> > Can some one please explain!
> 
> install black by `pip install black`
> 
> then run your file through black by
> `black filename.py` black is basically a auto format rule used by this repo owner all codes must pass this format, if you have any more doubt please reply back ASAP ✌🏼

After executing black filename.py, It shows All done, After that should I do anything ?
Thanks for replying 😊!

**Comment by onlinejudge95** [comment]
@saikumar-ramavath Let's fix the `pre-commit` check first.
This check is failing at `Run mypy .` [build log](https://github.com/TheAlgorithms/Python/pull/4503/checks?check_run_id=2814291017)

File: sorts/topological_sort.py [hunk 1]
```diff
@@ -8,28 +8,39 @@
 edges = {"a": ["c", "b"], "b": ["d", "e"], "c": [], "d": [], "e": []}
 vertices = ["a", "b", "c", "d", "e"]
 
+sort = []
+visited = {}
+
 
 def topological_sort(start, visited, sort):
```
Comment (onlinejudge95) [inline sorts/topological_sort.py:15]: Please provide type hints

File: sorts/topological_sort.py [hunk 2]
```diff
@@ -8,28 +8,39 @@
...
-    # if all vertices haven't been visited select a new one to visit
-    if len(visited) != len(vertices):
-        for vertice in vertices:
-            if vertice not in visited:
-                sort = topological_sort(vertice, visited, sort)
+    sort.append(current)    
     # return sort
     return sort
 
+def print_graph(sort):
```
Comment (onlinejudge95) [inline sorts/topological_sort.py:30]: This method is not required.

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

File: sorts/topological_sort.py [hunk 3]
```diff
@@ -8,28 +8,39 @@
...
+    #reverse the sort 
+    sort = sort[::-1]
+    print(sort)
```
Comment (Prajwalprakash3722) [inline sorts/topological_sort.py:33]: Try to avoid including print statements

**Review by Prajwalprakash3722 — CHANGES_REQUESTED** [review]

File: sorts/topological_sort.py (hunk 2 above)
Comment (saikumar-ramavath) [reply sorts/topological_sort.py]: Changes are made!
Can u please review it!

File: sorts/topological_sort.py (hunk 3 above)
Comment (saikumar-ramavath) [reply sorts/topological_sort.py]: Changes are made!
Can u please review it!

File: sorts/topological_sort.py [hunk 4]
```diff
@@ -8,28 +8,39 @@
...
+    # 1 value indicates that vertex is visited
+    for vertex in vertices:
+        visited.update({vertex: 0})
+
+    for vertex in visited:
+        if visited[vertex] == 0:
+            sort = topological_sort(vertex, visited, sort)
+
+    # reverse the sort
+    sort = sort[::-1]
```
Comment (onlinejudge95) [inline sorts/topological_sort.py:45]: Use `doctest` for testing

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

File: sorts/topological_sort.py (hunk 2 above)
Comment (onlinejudge95) [reply sorts/topological_sort.py]: You have still type hints changes remaining

File: sorts/topological_sort.py (hunk 4 above)
Comment (saikumar-ramavath) [reply sorts/topological_sort.py:45]: Type hints are added!
I'm a beginner, I'm not aware of doc testing
Thanks for helping me out 😊

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]
Please take a look at build info and fix it :)

**Pull request closed without merging**